dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/pages/project/WorkFlowDefinitionPage.java

```diff
@@ -0,0 +1,207 @@
+        taskRelationRequestData.conditionParams(conditionParams);
+        ArrayList<Object> taskRelationRequestDataList = new ArrayList<>();
+        taskRelationRequestDataList.add(taskRelationRequestData);
+
+        HashMap<String, Object> locations = new HashMap();
+        locations.put("taskCode", genNumId);
+        locations.put("x", 33.5);
+        locations.put("y", 38.5);
+        ArrayList<Object> locationsList = new ArrayList<>();
+        locationsList.add(locations);
+
+        workFlowCreateRequestData.locations(JSONUtils.toJsonString(locationsList));
+        workFlowCreateRequestData.taskDefinitionJson(JSONUtils.toJsonString(taskDefinitionRequestDataList));
+        workFlowCreateRequestData.taskRelationJson(JSONUtils.toJsonString(taskRelationRequestDataList));
+        workFlowCreateRequestData.name(workFlowName);
+        workFlowCreateRequestData.tenantCode(TenantAPITest.tenantName);
+        workFlowCreateRequestData.executionType(executionType);
+        workFlowCreateRequestData.description("");
+        workFlowCreateRequestData.globalParams(globalParams);
+        workFlowCreateRequestData.timeout(0);
+
+        headers.put(Constants.SESSION_ID_KEY, sessionId);
+        RequestClient requestClient = new RequestClient();
+        ProjectPage project = new ProjectPage();
+        String projectCode = project.getProjectCode(sessionId, projectName);
+
+        HttpResponse res = requestClient.post("/projects/" + projectCode + "/process-definition", headers, JSONUtils.convertValue(workFlowCreateRequestData, Map.class));
+        return res;
+    }
+
+    public HttpResponse queryWorkflow(String sessionId, String projectName, String workFlowName) {
+        Map<String, Object> params = new HashMap<>();
+        Map<String, String> headers = new HashMap<>();
+        params.put("pageSize", 10);
+        params.put("pageNo", 1);
+        params.put("searchVal", workFlowName);
+        headers.put(Constants.SESSION_ID_KEY, sessionId);
+        RequestClient requestClient = new RequestClient();
+        ProjectPage project = new ProjectPage();
+        String projectCode = project.getProjectCode(sessionId, projectName);
+        HttpResponse res = requestClient.get("/projects/" + projectCode + "/process-definition", headers, params);
+
+        for (WorkFlowResponseTotalList workFlowList : JSONUtils.convertValue(res.body().data(), WorkFlowResponseData.class).totalList()) {
+            workFlowCode = workFlowList.code();
+        }
+
+        return res;
+    }
+
+    public HttpResponse onLineWorkflow(String sessionId, String projectName, String workFlowName) {
+        WorkFlowDefinitionPage workflow = new WorkFlowDefinitionPage();
+        workflow.queryWorkflow(sessionId, projectName, workFlowName);
+        Map<String, Object> params = new HashMap<>();
+        Map<String, String> headers = new HashMap<>();
+        params.put("name", workFlowName);
+        params.put("releaseState", "ONLINE");
```

Comment:
It's better to take this parameter as the input parameter of the method, and the on and off lines are implemented in this method.